pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2380,6 +2398,16 @@ public boolean isSingleTenantCluster() {
     return endpointToInstance;
   }
 
+  public IdealState getLeadControllerResourceIdealState() {
```

Comment:
Do we need this two methods in the resource manager?

Reply:
The method `getLeadControllerResourceIdealState` is to check whether this resource is enabled or not. `getLeadControllerExternalView` is to fetch the master controller of partition, which will be used for realtime segment completion.

Reply:
I understand. My question is, should we keep these two methods in the resource manager (IMO they are not very related to the resource manager), or directly use helix admin to fetch them (same as table resource and broker resource.

Reply:
Right now `helix admin` isn't exposed to controller starter yet. If we choose the 2nd plan, it needs to expose to controller starter. Or always call `pinotHelixResourceManager.getHelixAdmin()` to get it.

Reply:
I prefer calling `pinotHelixResourceManager.getHelixAdmin()`, but you can decide